input/docs/handbook/testing/index.md

```diff
@@ -1,28 +1,108 @@
-# Testing
-
-ReactiveUI includes a few tools to help testing, building on what Rx already includes (see [introtorx](http://www.introtorx.com/content/v1.0.10621.0/16_TestingRx.html)) for details.
-
-The utilities are included in the ReactiveUI.Testing nuget package.
+ReactiveUI includes a few tools to help testing, built on what Reactive Extensions for .NET already include. The utilities are included in the `ReactiveUI.Testing` NuGet package. Make sure to install it into your unit tests project.
 
 ## Custom Scheduler
 
-A common pattern is to replace the schedulers ReactiveUI is using to test execution is occuring without injecting schedulers everywhere. This can be achieved in a couple of ways
+> Scheduling, and therefore threading, is generally avoided in test scenarios as it can introduce race conditions which may lead to non-deterministic tests — [Intro to Rx: Testing Rx](http://introtorx.com/Content/v1.0.10621.0/16_TestingRx.html)
+
+By default, `ReactiveCommand` uses `RxApp.MainThreadScheduler` and `ObservableAsPropertyHelper` uses `CurrentThreadScheduler.Instance`, but you can easily override this behavior:
 
-```csharp
-using(TestUtils.WithScheduler(ImmediateScheduler.Instance))
+```cs
+// Inject custom schedulers into your view model.
+public LoginViewModel(IScheduler customScheduler)
```

Comment:
I would potentially do customScheduler = null with a default value, similar to how we use Splat injection.

Reply:
Do you mean:
```cs
public LoginViewModel(IScheduler scheduler) {
  scheduler = scheduler ?? Locator.Current.GetService<IScheduler>();
}
```
Or fallback to default main thread scheduler:
```cs
public LoginViewModel(IScheduler scheduler) {
  scheduler = scheduler ?? RxApp.MainThreadScheduler;
}
```
Which option is better?

Reply:
I was thinking something like
```cs
public LoginViewModel(IScheduler scheduler = null) 
{
  scheduler = scheduler ?? Scheduler.Default; 
  scheduler = scheduler ?? RxApp.MainThreadScheduler;
}
```

The splat registration in this case is probably not what the user wants.